publication/2023-03-20_announcing_ngxs_3_8/article.md

```diff
@@ -0,0 +1,305 @@
+
+```ts
+import { LOCAL_STORAGE_ENGINE, SESSION_STORAGE_ENGINE } from '@ngxs/storage-plugin';
+@NgModule({
+  imports: [
+    NgxsStoragePluginModule.forRoot({
+      key: [
+        {
+          key: 'auth', // or `AuthState`
+          engine: LOCAL_STORAGE_ENGINE
+        },
+        {
+          key: UsersState, // or `users`
+          engine: SESSION_STORAGE_ENGINE
+        }
+      ]
+    })
+  ]
+})
+export class AppModule {}
+```
+
+Note that `LOCAL_STORAGE_ENGINE` and `SESSION_STORAGE_ENGINE` are provided by the storage plugin and will resolve to the `localStorage` and `sessionStorage`, respectively. When building SSR applications, these tokens will resolve to `null`, in this case a custom storage engine should be provided.
+
+### Form Plugin
+
+#### Feature: Allow ngxsFormDebounce to be string [#1972](https://github.com/ngxs/store/pull/1972)
+
+The ngxsFormDebounce is a static binding for now which means we're not watching its changes. We read it only once and provide to debounceTime. The change allows providing ngxsFormDebounce as a string, and not as a number so we extend possible values to be provided:
+
+```html
+// before
+<form [ngxsFormDebounce]="300">...</form>
+//after
+<form ngxsFormDebounce="300">...</form>
+```
+
+### Devtools Plugin
+
+#### Feature: New options added to `NgxsDevtoolsOptions` [#1879](https://github.com/ngxs/store/pull/1879)[#1968](https://github.com/ngxs/store/pull/1968)
+
+The devtools plugin now allows you to provide the following options:
+
+- `latency`: If more than one action is dispatched in the indicated interval, all new actions will be collected and sent at once.
+  It is the joint between performance and speed. When set to 0, all actions will be sent instantly.
+  Set it to a higher value when experiencing perf issues (also maxAge to a lower value). Default is 500 ms.
+- `actionsBlacklist`: string or array of strings as regex - actions types to be hidden in the monitors (while passed to the reducers).
+  If actionsWhitelist specified, actionsBlacklist is ignored.
+- `actionsWhitelist`: string or array of strings as regex - actions types to be shown in the monitors (while passed to the reducers).
+  If actionsWhitelist specified, actionsBlacklist is ignored.
+- `predicate`: called for every action before sending, takes state and action object, and
+  returns true in case it allows sending the current data to the monitor.
+  Use it as a more advanced version of actionsBlacklist/actionsWhitelist parameters
+- `trace`: if set to `true`, will include stack trace for every dispatched action, so you can see it in trace tab jumping directly to that part of code
+- `traceLimit`: maximum stack trace frames to be stored (in case trace option was provided as true)
+
+---
+
+## NGXS Labs Projects Updates
+
+### New Labs Project: @ngxs-labs/...
+
+...
+
+### Labs Project Updates: @ngxs-labs/...
+
+...
```

Comment:
`@ngxs-labs/firestore-plugin`

- Better compatibility with latest `firebase` and `@angular/fire`
Includes `NgxsFirestore` for modular and compat versions of `@angular/fire` to facilitate upgrade.
- New `cancelPrevious` option 'cancel-if-track-by-changed' to cancel previous connected action only when trackBy changed
- Support to fetch `metadata` fields (fromCache, hasPendingWrites) from Firestore